File: Backend-SpringBoot/FoodApp/src/main/java/com/tientien/foodapp/topping/entity/Topping.java
package com.tientien.foodapp.topping.entity;

import com.tientien.foodapp.common.entity.BaseEntity;
import com.tientien.foodapp.restaurant.entity.Restaurant;
import com.tientien.foodapp.topping.enums.ToppingStatus;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "toppings")
@Getter
@Setter
public class Topping extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Double price;

    private String description;

    private String image;

    @Enumerated(EnumType.STRING)
    private ToppingStatus status;

    @ManyToOne
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;
}

File: Frontend_Flutter/lib/models/topping.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'topping.freezed.dart';
part 'topping.g.dart';

@freezed
class Topping with _$Topping {
  const factory Topping({
    int? id,
    String? name,
    double? price,
    String? description,
    String? image,
    String? status,
    int? restaurantId,
    String? createdAt,
  }) = _Topping;

  factory Topping.fromJson(Map<String, dynamic> json) =>
      _$ToppingFromJson(json);
}
